packages/contracts-bedrock/src/L1/SuperchainConfig.sol

```diff
@@ -3,98 +3,169 @@ pragma solidity 0.8.15;
     event Paused(string identifier);
 
     /// @notice Emitted when the pause is lifted.
-    event Unpaused();
+    event Unpaused(string identifier);
 
     /// @notice Emitted when configuration is updated.
     /// @param updateType Type of update.
     /// @param data       Encoded update data.
     event ConfigUpdate(UpdateType indexed updateType, bytes data);
 
     /// @notice Semantic version.
-    /// @custom:semver 1.2.0
-    string public constant version = "1.2.0";
+    /// @custom:semver 2.0.0
+    string public constant version = "2.0.0";
 
     /// @notice Constructs the SuperchainConfig contract.
-    constructor() {
+    constructor() ReinitializableBase(2) {
         _disableInitializers();
     }
 
     /// @notice Initializer.
     /// @param _guardian    Address of the guardian, can pause the OptimismPortal.
-    /// @param _paused      Initial paused status.
-    function initialize(address _guardian, bool _paused) external initializer {
+    function initialize(address _guardian) external reinitializer(initVersion()) {
         _setGuardian(_guardian);
-        if (_paused) {
-            _pause("Initializer paused");
+    }
+
+    /// @notice Upgrades the SuperchainConfig contract.
+    function upgrade() external reinitializer(initVersion()) {
+        // Transfer the guardian into the new variable and clear the old storage slot.
+        bytes32 guardianSlot = bytes32(uint256(keccak256("superchainConfig.guardian")) - 1);
+        _setGuardian(Storage.getAddress(guardianSlot));
+        Storage.setBytes32(guardianSlot, bytes32(0));
+
+        // Clear the old paused slot.
+        // Note that if the pause was active while the upgrade was happening, the system will no
+        // longer be paused after the upgrade. Upgrades should generally not ever be executed while
+        // the system is paused, but it's worth noting that this is the case.
+        bytes32 pausedSlot = bytes32(uint256(keccak256("superchainConfig.paused")) - 1);
+        Storage.setBytes32(pausedSlot, bytes32(0));
+    }
+
+    /// @notice Returns the duration after which a pause expires.
+    /// @return The duration after which a pause expires.
+    function pauseExpiry() external pure returns (uint256) {
+        return PAUSE_EXPIRY;
+    }
+
+    /// @notice Pauses the system for a specific identifier.
+    /// @param _identifier The address identifier for the pause.
+    function pause(address _identifier) external {
+        // Only the Guardian can pause the system.
+        if (msg.sender != guardian) {
+            revert SuperchainConfig_OnlyGuardian();
+        }
+
+        // Cannot pause if the identifier is already paused to prevent re-pausing without either
+        // unpausing, extending, or resetting the pause timestamp.
+        if (pauseTimestamps[_identifier] != 0) {
+            revert SuperchainConfig_AlreadyPaused(_identifier);
         }
```

Comment:
I think that this makes it impossible for the following scenario to occur (and I'm not sure if it's desirable or not):

1. `pause()` is activated
2. The pause runs out after 3 months
3. `pause()` is called again.

Step 3 is impossible because `unpause()` was never called to explicitly set the timestamp back to `0`.

Reply:
Additional nit, but seeing as there is a `pausable()` function below, I'd suggest this change:

```suggestion
        // Cannot pause if the identifier is already paused to prevent re-pausing without either
        // unpausing, extending, or resetting the pause timestamp.
        if (!pausable()) {
            revert SuperchainConfig_AlreadyPaused(_identifier);
        }
```

Reply:
we may also want to change the name of the error to `NotPausable()` depending on the outcome of this conversation.

Reply:
This is the intended functionality